Sources/TensorFlow/Optimizer.swift

```diff
@@ -355,3 +355,91 @@ public class AdaGrad<Model: Layer>: Optimizer
         update(&model.allDifferentiableVariables, along: direction)
     }
 }
+
+public class AdaDelta<Model: Layer>: Optimizer
+    where Model.AllDifferentiableVariables == Model.TangentVector {
+    public typealias Model = Model
+    /// The learning rate.
+    public var learningRate: Float
+    ///decay factor, corresponding to fraction of gradient to keep at each time step.
+    public var rho: Float
+    /// A small scalar added to the denominator to improve numerical stability.
+    public var epsilon: Float
+    /// The learning rate decay.
+    public var decay: Float
+    /// The current step.
+    public var step: Int = 0
+    public var accumulators: Model.AllDifferentiableVariables
+    public var accDelta: Model.AllDifferentiableVariables
```

Comment:
- Could you add a doc comment that explains this property?
- Avoid abbreviations in this property name since `accDelta` is not a word of art and may be confusing.